### PR TITLE
[#354] syncing with the new Atom DB

### LINKED ISSUE #354: Sync QE with the new Atom DB integrated to C++ underlying implementation

### PR DESCRIPTION
**Pull request by angeloprobst**
Closes #354 
Relates to https://github.com/singnet/das-atom-db/issues/247

## All tests passed on my dev env with a locally-updated OpenFaaS, including unit and integration tests.

### CODE REVIEW

File: tests/integration/test_iterators.py [hunk 1]
```diff
@@ -322,6 +322,7 @@ def _check_asserts(self, das: DistributedAtomSpace, iterator: CustomQuery):
             next(iterator)
         return handles
 
+    @pytest.mark.skip("Requires Mongo indices to be updated with the new custom attributes")
```
Comment (andre-senna) [inline tests/integration/test_iterators.py:325]: Please make sure there's a card to address this fix and reference the card repo/number in the skip message.

Comment (andre-senna) [reply tests/integration/test_iterators.py:325]: Probably I missed this detail in the tests that have been skipped in the atomdb PR https://github.com/singnet/das-atom-db/pull/248 . Please fix the messages there as well.

**Review by andre-senna — APPROVED** [review]

**Review by eddiebrissow — APPROVED** [review]

File: hyperon_das/query_engines/query_engine_protocol.py [hunk 2]
```diff
@@ -130,9 +130,7 @@ def query(
         raise NotImplementedError
 
     @abstractmethod
-    def custom_query(
-        self, index_id: str, query: Query, **kwargs
-    ) -> Union[Iterator, List[Dict[str, Any]]]:
+    def custom_query(self, index_id: str, query: Query, **kwargs) -> Union[Iterator, List[AtomT]]:
```
Comment (marcocapozzoli) [inline hyperon_das/query_engines/query_engine_protocol.py:133]: This type hint shouldn't be `Iterator | list[AtomT]`

**Review by marcocapozzoli — APPROVED** [review]
I think the user guide should be changed with this new usage of NoteT and LinkT in add_node() and add_link()

File: hyperon_das/query_engines/query_engine_protocol.py (hunk 2 above)
Comment (angeloprobst) [reply hyperon_das/query_engines/query_engine_protocol.py:133]: Done!

**Comment by angeloprobst** [comment]
> I think the user guide should be changed with this new usage of NoteT and LinkT in add_node() and add_link()

I presume that the user guide will be completely revisited once we finish all refactorings that are currently in progress. @andre-senna ?

**Comment by andre-senna** [comment]
> > I think the user guide should be changed with this new usage of NoteT and LinkT in add_node() and add_link()
> 
> I presume that the user guide will be completely revisited once we finish all refactorings that are currently in progress. @andre-senna ?

Right.

**Pull request merged**